docs/en/latest/plugins/datadog.md

```diff
@@ -0,0 +1,131 @@
+
+This plugin pushes its custom metrics to the DogStatsD server, comes bundled with Datadog agent (to learn more about how to install a datadog agent, please visit [here](https://docs.datadoghq.com/agent/) ), over the UDP protocol. DogStatsD basically is an implementation of StatsD protocol which collects the custom metrics for Apache APISIX agent, aggregates it into a single data point and sends it to the configured Datadog server.
+To learn more about DogStatsD, please visit [DogStatsD](https://docs.datadoghq.com/developers/dogstatsd/?tab=hostagent) documentation.
+
+This plugin provides the ability to push metrics as a batch to the external Datadog agent, reusing the same datagram socket. In case if you did not receive the log data, don't worry give it some time. It will automatically send the logs after the timer function expires in our Batch Processor.
+
+For more info on Batch-Processor in Apache APISIX please refer.
+[Batch-Processor](../batch-processor.md)
+
+## Attributes
+
+| Name             | Type   | Requirement  | Default      | Valid | Description                                                                                |
+| -----------      | ------ | -----------  | -------      | ----- | ------------------------------------------------------------                               |
+| batch_max_size   | integer | optional    | 5000         | [1,...] | Max buffer size of each batch                                                            |
+| inactive_timeout | integer | optional    | 5            | [1,...] | Maximum age in seconds when the buffer will be flushed if inactive                       |
+| buffer_duration  | integer | optional    | 60           | [1,...] | Maximum age in seconds of the oldest entry in a batch before the batch must be processed |
+| max_retry_count  | integer | optional    | 1            | [1,...] | Maximum number of retries if one entry fails to reach dogstatsd server                   |
+
+## Metadata
+
+| Name        | Type    | Requirement |     Default        | Valid         | Description                                                            |
+| ----------- | ------  | ----------- |      -------       | -----         | ---------------------------------------------------------------------- |
+| host        | string  | optional    |  "127.0.0.1"       |               | The DogStatsD server host address                                      |
+| port        | integer | optional    |    8125            |               | The DogStatsD server host port                                         |
+| namespace   | string  | optional    |    "apisix"        |               | Prefix for all the custom metrics sent by APISIX agent. Useful for finding entities for metric graph. e.g. (apisix.request.counter)                                        |
```

Comment:
@bisakhmondal please add doc and examples for how to configure `host` and `port`, thanks

Reply:
I means how to set the datadog endpoint

Reply:
Yes, take this as an example: If I'm a newbie to use Apache APISIX and the datadog plugin, I'd like to know all the necessary steps after Apache APISIX runs successfully.

Like when I should update its metadata, and how I should setup its data with a Route.

Reply:
Got your point. Thanks for the elaborate explanation.
The plugin has a default metadata schema that assumes dogstatsd is available at localhost:8125. If someone needs to reconfigure that I have covered that on the blog which is soon going to be exported at apisix blog
Here is the link: https://blog.bisakh.com/blog/Monitoring-with-Datadog-in-Apache-APISIX

Do you think, we should add it here (the `Steps to Run Datadog Agent` section & `Custom Configuration`)? Thanks guys

Reply:
Ya, I have read your blog[1], I think the `Custom Configuration` part is what @moonming and I need 😄

![image](https://user-images.githubusercontent.com/2106987/141082091-0921c8c7-876c-4393-bc79-2e17482c4944.png)

[1] https://blog.bisakh.com/blog/Monitoring-with-Datadog-in-Apache-APISIX

Reply:
Cool Cool, I'll open a PR with the code snippet and some extra details. Thanks, @juzhiyuan @moonming